### PR TITLE
Session caching and cleanup

### PR DESCRIPTION
**Pull request by benbart**

### CODE REVIEW

File: rain_api_core/aws_util.py [hunk 1]
```diff
@@ -8,13 +8,16 @@
...
 from boto3 import client as botoclient, resource as botoresource, session as botosession, Session as boto_Session
+from boto3.resources.base import ServiceResource
 from botocore.config import Config as bc_Config
 from botocore.exceptions import ClientError
 
 log = logging.getLogger(__name__)
 secret_cache = {}
 region_list_cache = []
 region = ''
+botosess = None
```
Comment (bbuechler) [inline rain_api_core/aws_util.py]: Why not just do `botosess = botosession.Session()` here and skip the `if not botosess:` below?

Comment (benbart) [reply rain_api_core/aws_util.py]: I'm not entirely confident that Session() will execute properly outside of a function

Comment (bbuechler) [reply rain_api_core/aws_util.py]: We trust log will be there.. I don't think theres any reason to trust botosession any less!

Comment (benbart) [reply rain_api_core/aws_util.py]: already changed

**Pull request merged**